crates/vectorscan-sys/build.rs

```diff
@@ -0,0 +1,173 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License, Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+use std::env;
+use std::ffi::OsStr;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+/// Information about a project to compile with CMake
+struct Dependency {
+    /// The folder containing the extracted source of this dependency
+    source_path: PathBuf,
+    /// The path to the source tarball
+    tarball_path: PathBuf,
+}
+
+impl Dependency {
+    pub fn new<T: AsRef<OsStr>>(pretty_name: T, tarball_name: T) -> Self {
+        Self {
+            source_path: PathBuf::from(env::var("OUT_DIR").unwrap())
+                .join("vendored-sources")
+                .join(pretty_name.as_ref()),
+            tarball_path: Path::new(env!("CARGO_MANIFEST_DIR"))
+                .join("vendor")
+                .join(tarball_name.as_ref()),
+        }
+    }
+}
+
+fn main() {
+    let hs_dep = Dependency::new("vectorscan-5.4.11", "vectorscan-5.4.11.tar.gz");
+    let pcre_dep = Dependency::new("pcre-8.45", "pcre-8.45.tar.gz");
+
+    let dependencies = if cfg!(feature = "chimera") {
+        vec![&hs_dep, &pcre_dep]
+    } else {
+        vec![&hs_dep]
+    };
+
+    for dependency in dependencies {
+        println!(
+            "cargo:rerun-if-changed={}",
+            dependency.tarball_path.display()
+        );
+        if !dependency.source_path.exists() {
+            assert!(Command::new("mkdir")
+                .args(["-p", &dependency.source_path.to_string_lossy()])
+                .status()
+                .is_ok_and(|exit_status| exit_status.success()));
+            assert!(Command::new("tar")
+                .args([
+                    "-xzf",
+                    &dependency.tarball_path.to_string_lossy(),
+                    "-C",
+                    &dependency.source_path.to_string_lossy(),
+                    "--strip-components=1"
+                ])
+                .status()
+                .is_ok_and(|exit_status| exit_status.success()));
+        }
+    }
+
+    match env::var("CARGO_CFG_TARGET_OS").unwrap().as_str() {
+        "windows" => panic!("Vectorscan cannot be compiled for os: windows"),
```

Comment:
Is this temporary or permanent? We need to make it work for WIndows (and have it available in the IDE, etc)